src/Playwright/Helpers/Driver.cs

```diff
@@ -37,7 +37,18 @@ internal static string GetExecutablePath()
             DirectoryInfo assemblyDirectory = new(AppContext.BaseDirectory);
             if (!assemblyDirectory.Exists || !File.Exists(Path.Combine(assemblyDirectory.FullName, "Microsoft.Playwright.dll")))
             {
-                var assemblyLocation = typeof(Playwright).Assembly.Location;
+                string assemblyLocation = null;
+#pragma warning disable SYSLIB0012 // 'Assembly.CodeBase' is obsolete: 'Assembly.CodeBase and Assembly.EscapedCodeBase are only included for .NET Framework compatibility.
+                var assemblyLocationFileUrl = typeof(Playwright).Assembly.CodeBase;
+#pragma warning restore SYSLIB0012 // 'Assembly.CodeBase' is obsolete: 'Assembly.CodeBase and Assembly.EscapedCodeBase are only included for .NET Framework compatibility.
+                if (!string.IsNullOrEmpty(assemblyLocationFileUrl))
+                {
+                    assemblyLocation = new Uri(assemblyLocationFileUrl).LocalPath;
+                }
+                else
+                {
+                    assemblyLocation = typeof(Playwright).Assembly.Location;
+                }
```

Comment:
```suggestion
                string assemblyLocation;
                var assembly = typeof(Playwright).Assembly;
#pragma warning disable SYSLIB0012 // 'Assembly.CodeBase' is obsolete: 'Assembly.CodeBase and Assembly.EscapedCodeBase are only included for .NET Framework compatibility.
                if (Uri.TryCreate(assembly.CodeBase, UriKind.Absolute, out var codeBase) && codeBase.IsFile)
#pragma warning restore SYSLIB0012 // 'Assembly.CodeBase' is obsolete: 'Assembly.CodeBase and Assembly.EscapedCodeBase are only included for .NET Framework compatibility.
                {
                    assemblyLocation = codeBase.LocalPath;
                }
                else
                {
                    assemblyLocation = assembly.Location;
                }
```

Slightly more paranoid version as `new Uri` could throw.
The same logic is also used by `Microsoft.Extensions.DependencyModel`: 
https://github.com/dotnet/runtime/blob/e55c908229e36f99a52745d4ee85316a0e8bb6a2/src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextLoader.cs#L161

But they are using `CodeBase` as a fallback instead and don't prefer it over `Assembly.Location` like here.
https://github.com/dotnet/runtime/blob/e55c908229e36f99a52745d4ee85316a0e8bb6a2/src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextLoader.cs#L130-L171